cranelift-simplejit/src/memory.rs

```diff
@@ -31,7 +31,7 @@ impl PtrLen {
         let page_size = region::page::size();
         let alloc_size = round_up_to_page_size(size, page_size);
         unsafe {
-            let mut ptr: *mut libc::c_void = mem::uninitialized();
+            let mut ptr: *mut libc::c_void = mem::MaybeUninit::uninit().assume_init();
```

Comment:
This is wrong. Calling `assume_init` on an uninitialized `MaybeUninit` is UB.

```suggestion
            let mut ptr: *mut libc::c_void = 0 as *mut _;
```